### PR TITLE
Create an explanation label when hovering on the skip button

### PR DESCRIPTION
**Pull request by jbartka**

### CODE REVIEW

File: services/app/src/components/gameFooter/GameFooter.jsx [hunk 1]
```diff
@@ -18,9 +18,12 @@ function GameFooter({ skipCurrentPlayerMove }) {
         <button type="button" className="finish-btn" disabled={!lastTurn || isAllTurnsMade || (finished && !isAllTurnsMade)} onClick={finishMove}>
           Finish Move
         </button>
-        <button type="button" className="skip-btn" disabled={finished} onClick={() => skipCurrentPlayerMove()}>
-          Skip Move
-        </button>
+        <div className='hoverText'>
```
Comment (jvalatkevicius) [inline services/app/src/components/gameFooter/GameFooter.jsx:21]: Please maintain the class naming conventions by using `kebab-case` naming for class names, e.g. `hover-text`, `expl-text`, etc.

Comment (jvalatkevicius) [reply services/app/src/components/gameFooter/GameFooter.jsx:21]: Apart from that, the functionality we're building here is esentially a `tooltip` which is quite common. I think we should make it reusable.
The `Tooltip` component should be created. It should take a `title` as a prop and should be used like this:
```
<Tooltip title="Skips your move.">
  <button .../>
</Tooltip>
```

File: services/app/src/components/gameFooter/chess-pieces.css [hunk 2]
```diff
@@ -48,13 +48,40 @@
...
   background-color: transparent;
+  border: 1px solid black;
 }
 
 .skip-btn:disabled {
   color: rgba(60, 100, 177, 0.24);
   cursor: not-allowed;
 }
 
+.explText{
```
Comment (jvalatkevicius) [inline services/app/src/components/gameFooter/chess-pieces.css:59]: When working on `Tooltip` component, I suggest you look into this, to give you a basic idea https://www.w3schools.com/howto/howto_css_tooltip.asp

**Review by jvalatkevicius — CHANGES_REQUESTED** [review]

File: package.json [hunk 3]
```diff
@@ -34,6 +34,9 @@
     "lerna": "^4.0.0"
   },
   "dependencies": {
+    "@emotion/react": "^11.10.5",
+    "@emotion/styled": "^11.10.5",
+    "@mui/material": "^5.10.15",
```
Comment (jvalatkevicius) [inline package.json:39]: You included a couple of full ui libraries. Even though it provides a convenient set of components but it also adds quite a bit of weight to the project and there are a couple of alternatives to it. I think we need to agree as a team whether or not we really want to use this (or any other) UI library for the frontend, because it's quite a big decision to make.

If I'm not mistaken, we already had a discussion about using some sort of UI library in this project, however I cannot recall what was the agreement, so, please, raise this question in the slack channel and we'll agree on the decision.

**Review by jvalatkevicius — CHANGES_REQUESTED** [review]

File: services/app/src/components/gameFooter/GameFooter.jsx [hunk 4]
```diff
@@ -18,9 +20,11 @@ function GameFooter({ skipCurrentPlayerMove }) {
         <button type="button" className="finish-btn" disabled={!lastTurn || isAllTurnsMade || (finished && !isAllTurnsMade)} onClick={finishMove}>
           Finish Move
         </button>
-        <button type="button" className="skip-btn" disabled={finished} onClick={() => skipCurrentPlayerMove()}>
-          Skip Move
-        </button>
+        <Tooltip title="Skips your move. The game continues" className="tooltipClass" overlayInnerStyle={{fontSize: 15}} placement="rightTop" mouseEnterDelay = "0.001" mouseLeaveDelay= "0.001">
```
Comment (jvalatkevicius) [inline services/app/src/components/gameFooter/GameFooter.jsx:23]: I'd suggest to stick to the default enter and leave delays. 

Also, if you think we'll need to reuse some props in different places, we could move this Tooltip component into a wrapper Tooltip component that would pass these common props as defaults.

Apart from that, could we manage font size with the `tooltipClass` instead of passing inline styles?

File: package.json [hunk 5]
```diff
@@ -34,6 +34,9 @@
     "lerna": "^4.0.0"
   },
   "dependencies": {
+    "@emotion/react": "^11.10.5",
```
Comment (jvalatkevicius) [inline package.json:37]: Why is this needed?

File: package.json [hunk 6]
```diff
@@ -34,6 +34,9 @@
...
+    "@emotion/styled": "^11.10.5",
```
Comment (jvalatkevicius) [inline package.json:38]: Also this?

**Review by jvalatkevicius — CHANGES_REQUESTED** [review]

File: services/app/src/components/chessBoard/ExplanatoryTooltip.jsx [hunk 7]
```diff
@@ -0,0 +1,28 @@
...
+
+
+const ExplanatoryTooltip = ({
+    title,
+    children,
+    placement
+}) => {
+  return (
+    <Tooltip
+      className="tooltipClass"
```
Comment (jvalatkevicius) [inline services/app/src/components/chessBoard/ExplanatoryTooltip.jsx:13]: I could not find this class in this pull request, is it described somewhere?

File: services/app/src/components/gameFooter/chess-pieces.css [hunk 8]
```diff
@@ -0,0 +1,110 @@
+.flex-center {
```
Comment (jvalatkevicius) [inline services/app/src/components/gameFooter/chess-pieces.css:1]: what are all these classess?

Most of them does not seem to be used in this PR so are they needed or added accidentally?

**Review by jvalatkevicius — CHANGES_REQUESTED** [review]

**Review by jvalatkevicius — APPROVED** [review]

**Review by jvalatkevicius — APPROVED** [review]

**Review by jvalatkevicius — APPROVED** [review]

File: services/app/src/components/gameFooter/GameFooterButtons.jsx [hunk 9]
```diff
@@ -19,16 +19,16 @@ const GameFooterButtons = ({ skipCurrentPlayerMove }) => {
...
-          disabled={finished || isCurrentPlayerSpectator} 
-          onClick={() => skipCurrentPlayerMove()}
-        >
-          <ExplanatoryTooltip title="Skips your move. The game continues" placement="right">
-          Skip Move
-          </ExplanatoryTooltip>
-        </button>
+      <button 
+        type="button"
+        className="skip-btn font-size-s padding-x-m padding-y-s rubik-font"
```
Comment (jvalatkevicius) [inline services/app/src/components/gameFooter/GameFooterButtons.jsx:24]: why was this added?

**Review by jvalatkevicius — CHANGES_REQUESTED** [review]

File: services/app/src/components/gameFooter/GameFooterButtons.jsx (hunk 9 above)
Comment (AndzejIlj) [reply services/app/src/components/gameFooter/GameFooterButtons.jsx:24]: It came from main branch during the merge.
In the main branch: ```className=`"skip-btn font-size-s padding-x-m padding-y-s rubik-font" ```
Don't know why its highlighted as changed, but this line is the same as in main branch

Comment (jvalatkevicius) [reply services/app/src/components/gameFooter/GameFooterButtons.jsx:24]: Why is this class needed in the first place?

**Review by jvalatkevicius — APPROVED** [review]
